lib/openapi_parser/schema_validators/object_validator.rb

```diff
@@ -3,14 +3,17 @@ class ObjectValidator < Base
     # @param [Hash] value
     # @param [OpenAPIParser::Schemas::Schema] schema
     # @param [Boolean] parent_all_of true if component is nested under allOf
-    def coerce_and_validate(value, schema, parent_all_of: false)
+    # @param [String, nil] discriminator_property_name discriminator.property_name to ignore checking additional_properties
+    def coerce_and_validate(value, schema, parent_all_of: false, discriminator_property_name: nil)
       return OpenAPIParser::ValidateError.build_error_result(value, schema) unless value.kind_of?(Hash)
 
       return [value, nil] unless schema.properties
 
       required_set = schema.required ? schema.required.to_set : Set.new
       remaining_keys = value.keys
 
+      remaining_keys.delete(discriminator_property_name) if discriminator_property_name
```

Comment:
I think the property still needs to be defined the normal way? So the ```discriminator.property_name``` should be picking only out of existing properties?

Reply:
so e.g. if the property name is this: https://github.com/Ladas/openapi_parser/blob/40c766503b5f038ce35a02b377f83fcac7680932/spec/data/petstore-with-discriminator.yaml#L48

each object needs to define it, so e.g.
https://github.com/Ladas/openapi_parser/blob/40c766503b5f038ce35a02b377f83fcac7680932/spec/data/petstore-with-discriminator.yaml#L58

Reply:
Oh that's right, I will fix